lib/NotificationGenerator.php

```diff
@@ -52,7 +52,7 @@ public function sendNotificationForEvent(IEvent $event, int $activityId) {
 		// User is not the author or wants to see their own actions
 		$createStream = !$selfAction || $this->userSettings->getUserSetting($event->getAffectedUser(), 'setting', 'self');
 
-		if (($notifySetting || true) && $event->getGenerateNotification()) {
+		if ($notifySetting && $createStream && $event->getGenerateNotification()) {
```

Comment:
```suggestion
		if ($notifySetting && !$selfAction && $event->getGenerateNotification()) {
```

This line which is above can be removed (github diff just doesn't allow me to post that):
```php
$createStream = !$selfAction || $this->userSettings->getUserSetting($event->getAffectedUser(), 'setting', 'self');
```
The notification has nothing to do with `$createStream`

Also `'setting', 'self'` has been removed.

Actually what you would want is:

Reply:
> Also 'setting', 'self' has been removed.

Thanks, I missed that.